lotemplate/classes.py

```diff
@@ -321,6 +391,90 @@ def fill(self, variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
         :return: None
         """
 
+        def if_replace(doc, local_variables: dict[str, dict[str, Union[str, list[str]]]]) -> None:
+            """
+            Parse statements like [if $myvar==TOTO]...[endif]
+
+            If the condition matches we remove the if and endif statement.
+            It the condition doesn't match, we remove the statements and the text between the statements.
+
+            :param doc: the document to fill
+            :param local_variables: the variables
+            :return: None
+            """
+
+            def compute_if(local_x_found):
+                """
+                Compute the if statement.
+                """
+                if_statement = IfStatement(local_x_found.getString())
+                if_result = if_statement.get_if_result(local_variables[if_statement.variable_name]['value'])
+                if not if_result:
+                    # le if n'est pas vérifié => on efface le paragraphe avec le if
+                    text = local_x_found.getText()
+                    cursor = text.createTextCursorByRange(local_x_found)
+                    cursor.goLeft(len(if_statement.if_string), False)
+                    cursor.goRight(len(if_statement.if_string), True)
+                    if not cursor.goRight(1, True):
```

Comment:
This check shouldn't be in the `fill` method

Reply:
fixed